src/transformers/typescript.ts

```diff
@@ -249,24 +249,20 @@ function getCompilerOptions({
   options: Options.Typescript;
   basePath: string;
 }): CompilerOptions {
-  // default options
-  const compilerOptionsJSON = {
-    moduleResolution: 'node',
-    target: 'es6',
-  };
-
-  Object.assign(compilerOptionsJSON, options.compilerOptions);
+  const inputOptions = options.compilerOptions ?? {};
 
   const { errors, options: convertedCompilerOptions } =
     options.tsconfigFile !== false || options.tsconfigDirectory
-      ? loadTsconfig(compilerOptionsJSON, filename, options)
-      : ts.convertCompilerOptionsFromJson(compilerOptionsJSON, basePath);
+      ? loadTsconfig(inputOptions, filename, options)
+      : ts.convertCompilerOptionsFromJson(inputOptions, basePath);
 
   if (errors.length) {
     throw new Error(formatDiagnostics(errors, basePath));
   }
 
   const compilerOptions: CompilerOptions = {
+    target: ts.ScriptTarget.ES2021,
```

Comment:
Strictly speaking this is a breaking change for everyone who did not define a target before. I would change this to be es6 like before. Everyone who sets it otherwise is then able to alter it (which isn't possible currently because of the forced option).

Reply:
Ah yes, of course that makes sense.